cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/CruiseControlReconcilerTest.java

```diff
@@ -146,6 +153,20 @@ public void reconcileEnabledCruiseControl(VertxTestContext context) {
                     assertThat(cmCaptor.getAllValues().size(), is(1));
                     assertThat(cmCaptor.getValue(), is(notNullValue()));
 
+                    // Verify annotations
+                    CruiseControl expectCruiseControl = CruiseControl.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafka, VERSIONS, NODES, Map.of("kafka", kafka.getSpec().getKafka().getStorage()), Map.of(), SHARED_ENV_PROVIDER);
+                    ConfigMap configMap = expectCruiseControl.generateConfigMap(new MetricsAndLogging(null, null));
+
+                    Map<String, String> expectedAnnotations = new HashMap<>();
+                    expectedAnnotations.put(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_CERT_GENERATION, String.valueOf(clusterCa.caCertGeneration()));
+                    expectedAnnotations.put(Ca.ANNO_STRIMZI_IO_CLUSTER_CA_KEY_GENERATION, String.valueOf(clusterCa.caKeyGeneration()));
+                    expectedAnnotations.put(CruiseControl.ANNO_STRIMZI_SERVER_CONFIGURATION_HASH, Util.hashStub(configMap.getData().get(CruiseControl.SERVER_CONFIG_FILENAME)));
+                    expectedAnnotations.put(CruiseControl.ANNO_STRIMZI_CAPACITY_CONFIGURATION_HASH,  Util.hashStub(configMap.getData().get(CruiseControl.CAPACITY_CONFIG_FILENAME)));
+
+                    Map<String, String> actualAnnotations = depCaptor.getAllValues().get(0).getSpec().getTemplate().getMetadata().getAnnotations();
+                    assertThat("Annotations are not equal", actualAnnotations, is(expectedAnnotations));
```

Comment:
This is not the right way to do it. You should use the deployment captor to check what annotations you have in the Deployment that the operator created. You already seem to have the captor below, you just need to assert it.

Reply:
>  You should use the deployment captor to check what annotations you have in the Deployment that the operator created

I am not sure I understand, isn't that what the following line does?
```
Map<String, String> actualAnnotations = depCaptor.getAllValues().get(0).getSpec().getTemplate().getMetadata().getAnnotations();
```
uses the deployment captor, `depCaptor` to get a map of the annotations the operator created? Then compares against the annotation values we expect for the given configuration?

Or should I just use the deployment captor to check that required annotation keys exist in the deployment created by the operator?

Reply:
I guess it does -> but all the lines above it do not seem to make any sense. You should just check the annotation value is as expected. The hash should be stable. Also, it should likely be with the other deployment captor checks?

Reply:
I have refactored the code a little bit here, hopefully it is more clear. Let me know what you think!